### PR TITLE
kindletool: init at 1.6.5

### LINKED ISSUE #255935: Package request: Kindletool
**Project description**

 A tool for creating & extracting Kindle updates and more 

**Metadata**

* homepage URL: https://github.com/NiLuJe/KindleTool
* source URL: https://github.com/NiLuJe/KindleTool
* license: GPL3
* platforms: unix, linux

### PR DESCRIPTION
**Pull request by ocfox**
## Description of changes
close #255935 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ajs124 — COMMENTED** [review]
while it seems to build without it, their build systems might be happier if it had `pkg-config`.

also why is this called `kindle-tool` instead of `kindletool`?

**Comment by ocfox** [comment]
> also why is this called `kindle-tool` instead of `kindletool`?

By mistake, fixed.

File: pkgs/by-name/ki/kindletool/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  src = fetchFromGitHub {
+    owner = "NiLuJe";
+    repo = "KindleTool";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-Io+tfwgRAPEx+TQKZLBGrrHGAVS6ndgOOh+KlBh4t2U=";
+  };
+
+  patchPhase = ''
+    substituteInPlace KindleTool/Makefile \
+      --replace "DESTDIR?=/usr/local" "DESTDIR?=$out"
```
Comment (ajs124) [inline pkgs/by-name/ki/kindletool/package.nix:23]: Can't we set this somehow without using substitute? Maybe `env.DESTDIR` or in `cmakeFlags`?

Besides that, I think this is good to go.

Comment (ocfox) [reply pkgs/by-name/ki/kindletool/package.nix:23]: Tried but not work.

**Pull request closed without merging**